components/server/src/user/user-service.ts

```diff
@@ -118,16 +118,8 @@ export class UserService {
     }
 
     async setAdminRole(userId: string, targetUserId: string, admin: boolean): Promise<User> {
-        //TODO check if user has permission on targetUser to change admin role using auth system
-        const user = await this.userDb.findUserById(userId);
-        if (!user?.rolesOrPermissions || !user?.rolesOrPermissions.includes("admin")) {
-            throw new ApplicationError(ErrorCodes.PERMISSION_DENIED, "permission denied");
-        }
-
-        const target = await this.userDb.findUserById(targetUserId);
-        if (!target) {
-            throw new ApplicationError(ErrorCodes.NOT_FOUND, "not found");
-        }
+        await this.authorizer.checkPermissionOnUser(userId, "make_admin", targetUserId);
```

Comment:
Why can we already drop the old permissions system? We can't, right? Only after the migration ran.

Reply:
This method is not yet called from gitpod-server.impl and if it were the check would need to be there. The (new) services should only check based on the new FGA.

Reply:
So we have to keep in mind not to call this method until we rolled out the migration?

Reply:
You can call it from gitpod-server-impl if you have checked the permissions using the old system. This is the case for all functionality in services.